sunbird-rc-esignet-integration-impl/src/test/java/io/mosip/esignet/sunbirdrc/integration/service/SunbirdRCVCIssuancePluginTest.java

```diff
@@ -0,0 +1,297 @@
+
+        Map<String,Object> mockChallengMap=new HashMap<>();
+        Map<String,Object> credentialSubjectMap=new HashMap<>();
+        mockChallengMap.put("@context",List.of("https://www.w3.org/2018/credentials/examples/v1","https://www.w3.org/2018/credentials/v1"));
+        mockChallengMap.put("credentialSubject",credentialSubjectMap);
+        Mockito.when(objectMapper.readValue(Mockito.anyString(),Mockito.eq(Map.class))).thenReturn(mockChallengMap);
+
+        try{
+            sunbirdRCVCIssuancePlugin.getVerifiableCredentialWithLinkedDataProof(vcRequestDto,"holderId",identityMap);
+        }catch (VCIExchangeException e){
+            Assert.assertEquals(e.getErrorCode(), ErrorConstants.VCI_EXCHANGE_FAILED);
+        }
+    }
+
+    @Test
+    public void getVerifiableCredentialWithLinkedDataProof_InValidCredentialDetails_ThenFail() throws VCIExchangeException, JsonProcessingException {
+
+        VCRequestDto vcRequestDto=new VCRequestDto();
+        List<String> contextList=List.of("https://www.w3.org/2018/credentials/examples/v1","https://www.w3.org/2018/credentials/v1");
+        vcRequestDto.setContext(contextList);
+        vcRequestDto.setFormat("test");
+        List<String> types=new ArrayList<>();
+        types.add("VerifiableCredential");
+        types.add("InsurenceCredential");
+        vcRequestDto.setType(types);
+
+        Map<String,Object> identityMap=new HashMap<>();
+        identityMap.put("sub","osid");
+
+        Map<String,Object> responseMap=Map.of("policyNumber","654321","dob","654321");
+        ResponseEntity<Map<String,Object>>  responseEntity = new ResponseEntity(responseMap, HttpStatus.OK);
+        Mockito.when(restTemplate.exchange(
+                Mockito.any(RequestEntity.class),
+                Mockito.eq(new ParameterizedTypeReference<Map<String,Object>>() {}))).thenReturn(responseEntity);
+
+        Map<String,Object> mockChallengMap=new HashMap<>();
+        Map<String,Object> credentialSubjectMap=new HashMap<>();
+        mockChallengMap.put("@context",List.of("https://www.w3.org/2018/credentials/examples/v1","https://www.w3.org/2018/credentials/v1"));
+        mockChallengMap.put("credentialSubject",credentialSubjectMap);
+        Mockito.when(objectMapper.readValue(Mockito.anyString(),Mockito.eq(Map.class))).thenReturn(mockChallengMap);
+
+        try{
+            sunbirdRCVCIssuancePlugin.getVerifiableCredentialWithLinkedDataProof(vcRequestDto,"holderId",identityMap);
+        }catch (VCIExchangeException e){
+            Assert.assertEquals(e.getErrorCode(), ErrorConstants.VCI_EXCHANGE_FAILED);
+        }
+    }
+
+    @Test
+    public void validateAndLoadProperty_With_ValidDetails_ThenPass(){
+        Mockito.when(environment.getProperty(Mockito.anyString())).thenReturn("property");
+        ReflectionTestUtils.invokeMethod(sunbirdRCVCIssuancePlugin, "validateAndLoadProperty","propertyName","credentialProp",new HashMap<>());
```

Comment:
Can we avoid calling the private methods directly and rather call public methods like initialize to internally call and check the same logic ?

Reply:
same goes for below private methods also

Reply:
done